<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::WindowModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>766</width>
    <height>438</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>0</width>
    <height>0</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Smart RC - Control Pannel</string>
  </property>
  <widget class="QWidget" name="centralWidget">
   <widget class="QTabWidget" name="tabWidget">
    <property name="geometry">
     <rect>
      <x>20</x>
      <y>20</y>
      <width>731</width>
      <height>391</height>
     </rect>
    </property>
    <property name="currentIndex">
     <number>0</number>
    </property>
    <widget class="QWidget" name="tabEvents">
     <attribute name="title">
      <string>Events</string>
     </attribute>
     <widget class="QTableWidget" name="tblEvents">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>601</width>
        <height>291</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>601</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>601</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::SingleSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectRows</enum>
      </property>
      <attribute name="horizontalHeaderDefaultSectionSize">
       <number>80</number>
      </attribute>
      <attribute name="horizontalHeaderMinimumSectionSize">
       <number>39</number>
      </attribute>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <column>
       <property name="text">
        <string>No.</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Acitve</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Event Name</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Device Name</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Device ID</string>
       </property>
      </column>
     </widget>
     <widget class="QLabel" name="label">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Event list</string>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget">
      <property name="geometry">
       <rect>
        <x>630</x>
        <y>60</y>
        <width>81</width>
        <height>112</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout">
       <item>
        <widget class="QPushButton" name="btnAddEvent">
         <property name="text">
          <string>Add new</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnEdit">
         <property name="text">
          <string>Edit</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnDelEvent">
         <property name="text">
          <string>Delete</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnSaveEvents">
         <property name="text">
          <string>Save</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QLabel" name="label_8">
      <property name="geometry">
       <rect>
        <x>310</x>
        <y>330</y>
        <width>161</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>and by name (Event or Device):</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignRight|Qt::AlignTrailing|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QLineEdit" name="txtEventFilter">
      <property name="geometry">
       <rect>
        <x>480</x>
        <y>330</y>
        <width>131</width>
        <height>20</height>
       </rect>
      </property>
     </widget>
     <widget class="QLabel" name="label_9">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>330</y>
        <width>101</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Filter by event type:</string>
      </property>
      <property name="alignment">
       <set>Qt::AlignLeading|Qt::AlignLeft|Qt::AlignVCenter</set>
      </property>
     </widget>
     <widget class="QComboBox" name="cmbEventFilter">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>330</y>
        <width>181</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tabDevices">
     <attribute name="title">
      <string>Devices</string>
     </attribute>
     <widget class="QTableWidget" name="tblKnownDevices">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>30</y>
        <width>601</width>
        <height>291</height>
       </rect>
      </property>
      <property name="minimumSize">
       <size>
        <width>601</width>
        <height>0</height>
       </size>
      </property>
      <property name="maximumSize">
       <size>
        <width>601</width>
        <height>16777215</height>
       </size>
      </property>
      <property name="editTriggers">
       <set>QAbstractItemView::NoEditTriggers</set>
      </property>
      <property name="selectionMode">
       <enum>QAbstractItemView::SingleSelection</enum>
      </property>
      <property name="selectionBehavior">
       <enum>QAbstractItemView::SelectRows</enum>
      </property>
      <attribute name="horizontalHeaderStretchLastSection">
       <bool>true</bool>
      </attribute>
      <attribute name="verticalHeaderVisible">
       <bool>false</bool>
      </attribute>
      <column>
       <property name="text">
        <string>Device Name</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Status</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Device ID</string>
       </property>
      </column>
      <column>
       <property name="text">
        <string>Connect Address</string>
       </property>
      </column>
     </widget>
     <widget class="QLabel" name="label_2">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>10</y>
        <width>81</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Known devices</string>
      </property>
     </widget>
     <widget class="QWidget" name="layoutWidget_2">
      <property name="geometry">
       <rect>
        <x>630</x>
        <y>60</y>
        <width>81</width>
        <height>112</height>
       </rect>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout_2">
       <item>
        <widget class="QPushButton" name="btnDiscover">
         <property name="text">
          <string>Discover</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnDeviceProp">
         <property name="text">
          <string>Properties...</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnDelDevice">
         <property name="text">
          <string>Delete</string>
         </property>
        </widget>
       </item>
       <item>
        <widget class="QPushButton" name="btnSave">
         <property name="text">
          <string>Save</string>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QCommandLinkButton" name="cmdUpdateNow">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>320</y>
        <width>171</width>
        <height>31</height>
       </rect>
      </property>
      <property name="text">
       <string>Poll devices now</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_7">
      <property name="geometry">
       <rect>
        <x>190</x>
        <y>330</y>
        <width>131</width>
        <height>21</height>
       </rect>
      </property>
      <property name="text">
       <string>Filter by device type:</string>
      </property>
     </widget>
     <widget class="QComboBox" name="cmbDeviceFilter">
      <property name="geometry">
       <rect>
        <x>330</x>
        <y>330</y>
        <width>281</width>
        <height>22</height>
       </rect>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tab">
     <attribute name="title">
      <string>Execution</string>
     </attribute>
     <widget class="QWidget" name="horizontalLayoutWidget">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>60</y>
        <width>211</width>
        <height>41</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="lyBottom"/>
     </widget>
     <widget class="QPushButton" name="btnLog">
      <property name="geometry">
       <rect>
        <x>600</x>
        <y>20</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>View log</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_6">
      <property name="geometry">
       <rect>
        <x>10</x>
        <y>20</y>
        <width>101</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Event loop state: </string>
      </property>
     </widget>
     <widget class="QLabel" name="lblELState">
      <property name="geometry">
       <rect>
        <x>120</x>
        <y>20</y>
        <width>211</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Unknown</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnPing">
      <property name="geometry">
       <rect>
        <x>350</x>
        <y>20</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Ping</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnReload">
      <property name="geometry">
       <rect>
        <x>430</x>
        <y>20</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Reload</string>
      </property>
     </widget>
     <widget class="QPushButton" name="btnQuit">
      <property name="geometry">
       <rect>
        <x>510</x>
        <y>20</y>
        <width>75</width>
        <height>23</height>
       </rect>
      </property>
      <property name="text">
       <string>Quit!</string>
      </property>
     </widget>
    </widget>
    <widget class="QWidget" name="tabOptions">
     <attribute name="title">
      <string>Options</string>
     </attribute>
     <widget class="QSpinBox" name="spnPollTime">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>20</y>
        <width>42</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>5</number>
      </property>
      <property name="maximum">
       <number>60</number>
      </property>
      <property name="value">
       <number>5</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_3">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>20</y>
        <width>121</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Device polling time (min)</string>
      </property>
     </widget>
     <widget class="QLabel" name="label_4">
      <property name="geometry">
       <rect>
        <x>40</x>
        <y>50</y>
        <width>131</width>
        <height>16</height>
       </rect>
      </property>
      <property name="text">
       <string>Device request timeout(s)</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spnTimeOut">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>50</y>
        <width>42</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>1</number>
      </property>
      <property name="maximum">
       <number>60</number>
      </property>
      <property name="value">
       <number>5</number>
      </property>
     </widget>
     <widget class="QWidget" name="horizontalLayoutWidget_2">
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>110</y>
        <width>201</width>
        <height>31</height>
       </rect>
      </property>
      <layout class="QHBoxLayout" name="lySaveOnExit"/>
     </widget>
     <widget class="QPlainTextEdit" name="plainTextEdit">
      <property name="enabled">
       <bool>false</bool>
      </property>
      <property name="geometry">
       <rect>
        <x>30</x>
        <y>160</y>
        <width>661</width>
        <height>171</height>
       </rect>
      </property>
      <property name="plainText">
       <string>Thanx to:
Tamas Karpati
Pedro Correia
Iman Ahmadvand
Wei Dai and the Crypto++ Project
Ipsum Domus (blog.ipsumdomus.com)
Matthew Garrett</string>
      </property>
     </widget>
     <widget class="QSpinBox" name="spnMaxLog">
      <property name="geometry">
       <rect>
        <x>180</x>
        <y>80</y>
        <width>71</width>
        <height>22</height>
       </rect>
      </property>
      <property name="minimum">
       <number>10</number>
      </property>
      <property name="maximum">
       <number>2000</number>
      </property>
      <property name="singleStep">
       <number>100</number>
      </property>
      <property name="value">
       <number>100</number>
      </property>
     </widget>
     <widget class="QLabel" name="label_5">
      <property name="geometry">
       <rect>
        <x>20</x>
        <y>80</y>
        <width>151</width>
        <height>20</height>
       </rect>
      </property>
      <property name="text">
       <string>Maximum lines to log (2K max)</string>
      </property>
     </widget>
    </widget>
   </widget>
  </widget>
  <widget class="QToolBar" name="mainToolBar">
   <attribute name="toolBarArea">
    <enum>TopToolBarArea</enum>
   </attribute>
   <attribute name="toolBarBreak">
    <bool>false</bool>
   </attribute>
  </widget>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <resources/>
 <connections/>
</ui>
